index.js

```diff
@@ -43,7 +43,18 @@ module.exports = function (bodyParser) {
             return next(err);
           }
 
-          req.body = xml || req.body;
+          // Prevent setting __proto__ and constructor.prototype
+          const safe = {};
+          for (const key in xml) {
+            if (
+              key !== '__proto__' &&
+              key !== 'constructor' &&
+              key !== 'prototype'
+            ) {
+              safe[key] = xml[key];
+            }
+          }
+          req.body = safe || req.body;
```

Comment:
I think in this case it is fine to simply `delete` the offending keys.

```suggestion
          if(xml) {
            // Guard against prototype pollution
            delete xml.__proto__;
            delete xml.constructor;
            delete xml.prototype;
  
            req.body = xml;
          }
```